Feat/xxx/ifm barcode nodered

krgrsebastian opened the pull request:
Added retrofitting with ifm gateways, data connectivity with node-RED, and barcodereaders to the features section

krgrsebastian committed:
Ifm-retrofitting

krgrsebastian committed:
Ifm-retrofitting

krgrsebastian committed:
Ifm-retrofitting

krgrsebastian committed:
Ifm-retrofitting

krgrsebastian committed:
Ifm-retrofitting

krgrsebastian committed:
Data connectivity with Node-RED

krgrsebastian committed:
#

krgrsebastian committed:
#

krgrsebastian committed:
barcodereader

github-actions[bot] committed:
Merge main into feat/xxx/ifm_barcode_nodered

krgrsebastian committed:
barcodereader

krgrsebastian committed:
Merge remote-tracking branch 'origin/feat/xxx/ifm_barcode_nodered' into feat/xxx/ifm_barcode_nodered

krgrsebastian committed:
barcodereader

github-actions[bot] committed:
Merge main into feat/xxx/ifm_barcode_nodered

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -18,16 +24,27 @@ By default, you can
+- connect to kafka (node-red-contrib-kafkajs)
+- connect to Siemens-S7 (node-red-contrib-s7)
+- connect to serial (node-red-node-serialport)
+
```
There are many more: https://learn.umh.app/blog/node-red-in-industrial-iot-a-growing-standard/

https://flows.nodered.org/collection/U7bo_px1Nm6h
https://flows.nodered.org/node/node-red-contrib-cip-ethernet-ip

Also take a look in the industrial forum: https://discourse.nodered.org/c/industrial/18

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -18,16 +24,27 @@ By default, you can
 - connect to HTTP using the HTTP node
```
you can also connect to MQTT, TCP/IP and some others by default

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -18,16 +24,27 @@ By default, you can
 ## How can I use it?
 
-See also tutorial on accessing Node-RED
+Node-red comes preinstalled as a microservice with the umh. Follow our [get started](/docs/getstarted/) guide, 
```
Please also use either UMH or United Manufacturing Hub. Also check the spelling of Node-RED

JeremyTheocharis replied:
Do a step-by-step tutorial here.

1. Open Node-RED
2. Use default node such as HTTP or install the flow of your protocol (see above for an exemplery list
3. Process the data and align it with the UMH datamodel. Create e.g., process value messages (link to historian feature and process values). 
4. Send the result to MQTT / Kafka (link to tutorial)

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -18,16 +24,27 @@ By default, you can
+- most packages have no enterprise support. If you encounter any errors, you need to ask the community. 
+  However, we found that these packages are often more stable than the commercial ones out there, 
+  as they have been battle tested by way more users than commercial software.
+- having many flows without following a strict structure, leads to confusion for other users
```
good limitation. not only for other users, but in general.

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -18,16 +24,27 @@ By default, you can
 
-- most packages have no enterprise support. If you encounter any errors, you need to ask the community. However, we found that these packages are often more stable than the commercial ones out there, as they have been battle tested by way more users than commercial software
-- 
+- most packages have no enterprise support. If you encounter any errors, you need to ask the community. 
```
one additional limitation is "the speed of development of Node-RED". Klaus Landsdorf for example spends at least around 40h per year just to keep a node available and up to date. There are half-year significant updates to Node-RED, which might break old nodes. These update cycles put pressure on the IT / OT organization using this as they need to keep Node-RED updated

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
 +++
 
-Introduction sentence
+Retrofitting older machines with sensors is sometimes the only-way to capture process-relevant information.
```
Link to retroffitting article explaing the possible methods

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
-Introduction sentence
+Retrofitting older machines with sensors is sometimes the only-way to capture process-relevant information.
+In this article, we will focus on retrofitting with ifm IO-link gateways and Sensorconnect, a microservice of the 
+united manufacturing hub, that finds and reads out ifm IO-link masters in the network and pushes sensor data to 
```
Please check spelling of UMH

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
 
 ## How can I use it?
 
+To use ifm IO-link gateways and [Sensorconnect](/docs/architecture/microservices/core/sensorconnect/), you need to first
```
Do a step-by-step tutorial here

1. Connect the IO-link gateways to the same network
2. Retrofit the machine with sensors and connect them to the IO-link gateway
3. Configure the sensorconnect ip range to either match the IP address using subnet notation `/32`, or if you have multiple masters, configure it to scan an entire range `/24/
4. Data should appear in your Unified Namespace

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
 ## What are the limitations?
 
-- the current ifm firmware has a software bug, that will cause the IO-link master to crash if it receives to many requests. An experimental firmware is available upon request
+- the current ifm firmware has a software bug, that will cause the IO-link master to crash if it receives to many requests.
```
Fix are either the firmware, or to re-connect the hardware with power.

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
+Sensorconnect offers: 
+- automatic detection of ifm IO-link master in the network
+- identification of IO-Link or other digital sensors connected to the gateways
+- constant polling of data from the detected sensors with a high frequency
```
which frequency?

JeremyTheocharis commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
+- automatic detection of ifm IO-link master in the network
+- identification of IO-Link or other digital sensors connected to the gateways
+- constant polling of data from the detected sensors with a high frequency
+- interpreting the received data based on a sensor database containing thousands of entries
```
which database?

krgrsebastian committed:
feedback jeremy

krgrsebastian committed:
feedback jeremy

krgrsebastian committed:
feedback jeremy

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/united-manufacturing-hub/umh.docs.umh.app?pullRequest=82) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/united-manufacturing-hub/umh.docs.umh.app?pullRequest=82) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/united-manufacturing-hub/umh.docs.umh.app?pullRequest=82) it.</sub>

Commit:
fix: Corrected a few spelling mistakes
change: first sentence, less repetitive
add: Step by Step instruction for the barcode reader

Commit:
fix: Corrected a few spelling mistakes, rephrased stuff etc.
add:
-Step by Step instruction for sensorconnect
-Links to ifm and the retrofitting blog article
-info for analog and digital sensors

dmUMH commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,51 @@
+  An experimental firmware is available upon request
 
 ## Where to get more information?
+
```
ich hoffe ich habe den richtigen Link eingefügt.

Commit:
add: Link to the community repo

Commit:
fix: spelling

Commit:
fix: spelling

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
 +++
 
-Introduction sentence
+One feature of the UMH is to connect devices on the shopfloor such as PLCs, Quality Stations or MES / ERP systems with the Unified Namespace using Node-RED. 
```
Use United Manufacturing Hub instead of UMH

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
 
 ## When should I use it?
 
+Sometimes it is necessary to connect a lot of different protocols (e.g Siemens-S7, OPC-UA, Serial, ...) and node-RED can be a maintainable
```
Avoid latin phrases
https://umh.docs.umh.app/docs/development/contribute/documentation/style/style-guide/#avoid-latin-phrases

krgrsebastian replied:
dont get that point. I didnt use "We"  \_(*.*)_/

DaruZero replied:
I was talking about the "e.g", but it's too late now I guess

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
 
 By default, you can
 - connect to MQTT using the MQTT node
 - connect to HTTP using the HTTP node
+- connect to TCP using the TCP node
+- connect to IP using the UDP node
 
 By installing additional packages, you can:
-- connect to OPC-UA (node-red-controb-opcua)
-- ...
+- connect to OPC-UA (node-red-contrib-opcua)
+- connect to kafka (node-red-contrib-kafkajs)
+- connect to Siemens-S7 (node-red-contrib-s7)
+- connect to serial (node-red-node-serialport)
+- connect to modbus (node-red-contrib-modbus had)
+- connect to MC-protocol (node-red-contrib-mcprotocol)
+- connect to OMRON FINS Ethernet protocol (node-red-contrib-omron-fins)
+- connect to EtherNet/IP Protocol (node-red-contrib-cip-ethernet-ip)
+- connect to PostgreSQL (node-red-contrib-postgresql)
+- connect to SAP SQL Anywhere
```
Really ugly to read. Try something on the line of this:

By default there connector nodes for common protocols:
- MQTT
- HTTP
- TCP
- IP

Furthermore, you can install packages to support more connection protocols. For example:
- OPC-UA (node-red-contrib-opcua)
- ...

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
-See also tutorial on accessing Node-RED
+Node-RED comes preinstalled as a microservice with the United Manufacturing Hub.
+
+1. To access Node-RED navigate to **network**->**services** on the left-hand side in UMHLens. You can download UMHLens [here](https://github.com/united-manufacturing-hub/UMHLens/releases).
```
Capitalize Network and Services, and add a space before and after the arrow. `**Network** -> **Services**`
Use `resource` shortcode for UMHLens. `{{< resource type="lens" name="name" >}}`

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
+Node-RED comes preinstalled as a microservice with the United Manufacturing Hub.
+
+1. To access Node-RED navigate to **network**->**services** on the left-hand side in UMHLens. You can download UMHLens [here](https://github.com/united-manufacturing-hub/UMHLens/releases).
+2. Change the namespace to **Namespace: united-manufacturing-hub** at the top right.
```
On the top right, change the Namespace from default to {{< resource type="ns" name="umh" >}}

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
+
+   ![Untitled](/images/getstarted/managingTheSystem/getStartedManagingServices.png?width=75%)
+
+3. Click on **united-manufacturing-hub-nodered-service** and scroll down to **Connection** and forward the port.
```
Use `resource` shortcode for node red service. Use a comma instead of the firs "and"

DaruZero commented on umh.docs.umh.app/content/en/docs/features/data-connectivity-node-red/index.md:
```diff
@@ -3,31 +3,78 @@ title = "Data connectivity with Node-RED"
 
-- most packages have no enterprise support. If you encounter any errors, you need to ask the community. However, we found that these packages are often more stable than the commercial ones out there, as they have been battle tested by way more users than commercial software
-- 
+- most packages have no enterprise support. If you encounter any errors, you need to ask the community. 
+  However, we found that these packages are often more stable than the commercial ones out there, 
+  as they have been battle tested by way more users than commercial software.
+- having many flows without following a strict structure, leads in general to confusion.
+- one additional limitation is "the speed of development of Node-RED". After a big Node-RED and JavaScript update
+  dependencies most likely break, and those single community maintained nodes need to be updated.
```
Capitalize first letter.
Last point is a bit confusing, what do you want to say there?

DaruZero commented on umh.docs.umh.app/content/en/docs/features/ifm-retrofitting/index.md:
```diff
@@ -1,24 +1,54 @@
 
-Introduction sentence
+[Retrofitting](https://learn.umh.app/blog/connectivity-retrofitting-the-shopfloor-with-plug-play-sensors/) older machines with sensors is sometimes the only-way to capture process-relevant information.
+In this article, we will focus on retrofitting with [ifm IO-Link gateways](https://www.ifm.com/de/de/category/245) and Sensorconnect, a microservice of the 
```
Maybe link the first occurrence of Sensorconnect to its documentation page

DaruZero reviewed (changes requested):
Why are the new articles in page bundles if it only contains one `.md` file? Please convert to standalone pages.
From this
```
features/
├── barcodereader-retrofitting/
│   └── index.md
├── data-connectivity-node-red/
│   └── index.md
...
```
to this
```
features/
├── barcodereader-retrofitting.md
├── data-connectivity-node-red.md
...
```

Also. please  use the correct formatting style. 
- Each line should not be longer than 80 characters. You can set a vertical line in your text editor to help you with that. Depending on the editor you use, the setting might be called ruler, right margin, or column
- Use a markdown linter. For example, in VS code there is an extension called markdownlint that will show you common style errors, like not having a blank line before and after a code block ` ``` ... ``` `

Lastly, remember to squash the commits before merging. [Here](https://umh.docs.umh.app/docs/development/contribute/new-content/github-workflow/#squash-commits) more info about that. Ask me for help if you have doubts

JeremyTheocharis commented:
> Why are the new articles in page bundles if it only contains one `.md` file? Please convert to standalone pages. From this
> 
> ```
> features/
> ├── barcodereader-retrofitting/
> │   └── index.m
> ├── data-connectivity-node-red/
> │   └── index.md
> ...
> ```
> 
> to this
> 
> ```
> features/
> ├── barcodereader-retrofitting.md
> ├── data-connectivity-node-red.md
> ...
> ```

Because the original template from me had this structure as well. If we change that structure, we should either change all of the mand introduce redirects. Or we just leave it like this.

DaruZero commented:
Ok, then maybe in another PR we should change the structure, and maybe also add the archetype (already filed an issue for that https://github.com/united-manufacturing-hub/umh.docs.umh.app/issues/124)

krgrsebastian committed:
included Matteos Feedback
